### PR TITLE
internal: refactor storage credentials

### PR DESCRIPTION
**Pull request by daniel-weisse**
### Proposed change(s)

- Refactor KMS storage backends to use credentials in a similar way to the KMS backends.
- Add storage URI parsing functions

### Checklist

- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [ ] Add labels (e.g., for changelog category)
- [ ] Link to Milestone

### CODE REVIEW

File: internal/kms/uri/uri_test.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	}
+
+	checkURI(t, cfg, DecodeGoogleCloudStorageConfigFromURI)
+}
+
+type cfgStruct interface {
+	EncodeToURI() string
+}
+
+func checkURI[T any](t *testing.T, cfg cfgStruct, decodeFunc func(string) (T, error)) {
```
Comment (derpsteb) [inline internal/kms/uri/uri_test.go:103]: *insertfeelsgoodmanmeme*

File: internal/kms/README.md [hunk 2]
```diff
@@ -74,109 +75,30 @@ Supported are:
...
-
-The client will use the specified Blob container if it already exists, or create it first otherwise.
-
-To create the client:
-
-```Go
-connectionString := "DefaultEndpointsProtocol=https;AccountName=<myAccountName>;AccountKey=<myAccountKey>;EndpointSuffix=core.windows.net"
-store, err := storage.NewAzureStorage(context.TODO(), connectionString, "myContainer", nil)
-```
+Authorization for Azure Key Vault happens through the use of manged identities.
```
Comment (derpsteb) [inline internal/kms/README.md:85]:
```suggestion
Authorization for Azure Blob Storage happens through the use of manged identities.
```

File: internal/kms/README.md [hunk 3]
```diff
@@ -74,109 +75,30 @@ Supported are:
...
+The managed identity requires the following permissions:
 
-### Google Cloud Storage
+* `Microsoft.Storage/storageAccounts/blobServices/containers/write` (only if a storage container is not set up in advance)
+* `Microsoft.Storage/storageAccounts/blobServices/containers/blobs/write`
+* `Microsoft.Storage/storageAccounts/blobServices/containers/blobs/read`
+* `Microsoft.Storage/storageAccounts/blobServices/containers/blobs/add/action`
```
Comment (derpsteb) [inline internal/kms/README.md:91]: You also need these on the storage account you are using:
- Storage Blob Data Contributor
- Storage Queue Data Contributor

**Review by derpsteb — APPROVED** [review]
Good stuff! After the fixes manual testing seemed to work as expected.

**Comment by derpsteb** [comment]
Ah if you could hold off with merging until this one is done, that would be nice:
https://github.com/edgelesssys/constellation/actions/runs/4313222953

File: internal/kms/README.md (hunk 3 above)
Comment (daniel-weisse) [reply internal/kms/README.md:91]: I'm not 100% sure about how exactly Azure IAM works, but I think `Storage Blob Data Contributor` and `Storage Queue Data Contributor` are roles that provide permissions to access storage accounts, e.g. `Data Contributor` provides the `Microsoft.Storage/storageAccounts/blobServices/containers/blobs/write` permission

**Pull request merged**